crates/cext/src/transpiler/passes/optimize_1q_decomposition.rs

```diff
@@ -0,0 +1,103 @@
+
+/// @ingroup QkTranspilerPasses
+/// Runs the Optimize1qGatesDecomposition pass in standalone mode on a circuit.
+///
+/// Optimize1qGatesDecomposition, as its name implies, optimizes chains of single-qubit
+/// gates by combining them into a single gate.
+///
+/// The decision of whether to replace the original chain depends on:
+/// - If the original chain was out of basis.
+/// - If the original chain was in basis but the replacement has lower error rates.
+/// - If the original chain is an identity (chain gets removed).
+///
+/// The error is the combined multiplication of the errors of individual gates on the
+/// qubit it operates on.
+///
+/// @param circuit A pointer to the ``QkCircuit`` object to transform.
+/// @param target A pointer to the ``QkTarget`` object or a null pointer.
+/// In the case a null pointer is provided the pass calculates the error
+/// of each instruction based on each one-qubit sequence's length, and
+/// will support all basis gates on its Euler basis set.
+///
+/// @return The circuit after applying the optimizations.
+///
+/// # Example
+///
+///     QkTarget *target = qk_target_new(1);
+///     double u_errors[3] = {0., 1e-4, 1e-4};
+///     for (int idx = 0; idx < 3; idx++) {
+///         QkTargetEntry *u_entry = qk_target_entry_new(QkGate_U);
+///         uint32_t qargs[1] = {
+///             0,
+///         };
+///         qk_target_entry_add_property(u_entry, qargs, 1, NAN, u_errors[idx]);
+///         qk_target_add_instruction(target, u_entry);
+///     }
+///
+///     // Build circuit
+///     QkCircuit *circuit = qk_circuit_new(1, 0);
+///     uint32_t qubits[1] = {0};
+///     for (int iter = 0; iter < 3; iter++) {
+///         qk_circuit_gate(circuit, QkGate_H, qubits, NULL);
+///     }
+///
+///     // Run transpiler pass
+///     qk_transpiler_standalone_optimize_1q_gates_decomposition(circuit, target);
+///
+///     // Clean up
+///     qk_target_free(target);
+///     qk_circuit_free(circuit);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to a ``QkCircuit`` and
+/// if ``target`` is not a valid pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_transpiler_standalone_optimize_1q_gates_decomposition(
+    circuit: *mut CircuitData,
+    target: *const Target,
+) {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe {
+        if target.is_null() {
+            None
+        } else {
+            Some(const_ptr_as_ref(target))
+        }
+    };
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let circuit = unsafe { mut_ptr_as_ref(circuit) };
+
+    // Convert the circuit to a DAG.
+    let mut circuit_as_dag = DAGCircuit::from_circuit_data(circuit, false, None, None, None, None)
+        .expect("Error while converting the circuit to a dag.");
```

Comment:
It would be nice if we would have a set of `TranspilerError`s which we could return here instead of just panicking... but maybe this would need to be done generally for all passes in a follow-up.